www/js/metrics/MetricsDateSelect.tsx

```diff
@@ -35,10 +35,18 @@ const MetricsDateSelect = ({ dateRange, setDateRange }: Props) => {
 
   const onChoose = useCallback(
     ({ startDate, endDate }) => {
+      const dtStartDate = DateTime.fromJSDate(startDate).startOf('day')
+      let dtEndDate;
+      if(!endDate) { // For when only one day is selected
+        dtEndDate = dtStartDate.endOf('day').minus({minutes: 1});
```

Comment:
Is there a good reason to use the endOf the day here when usually we just use the startOf the day?
If we used startOf the day for both ends of the selected range, is the behavior any different?

I am not sure that the server cares what part of the day the request timestamps refer to since it chunks into full days anyway -- but let me know if you noticed otherwise.
If there is no difference, this code can be simplified a lot

Reply:
Gosh, thank you for asking -- This is going to be a wall of text, but TL;DR, I think this is a lot more subtle of an issue than I originally thought!

First, I think we probably don't need the `minus()` -- I remember it making a difference and decided to keep it, but can't seem to replicate that again... I'm thinking I'll test a little more, and axe it if I don't see a difference.

If we set the `endDate` to the `startDate.startOf()`, the server query returns blank. I had the same idea, since (irrc) BEMUserCache queries work fine with a single timestamp.  From what understand, the `BEMServerComm` to query to "result/metrics/${}"  in commHelper needs the times to be different.  I haven't looked too far into it, but will do so after writing this up.

Interestingly, this brings me to another issue I just found as I was writing this -- there actually _is_ a way to select a single date range, without encountering the `null` value.  This is done, somewhat unintuitively, by double tapping given date.  But, because both values are the start-date, this  shows up empty as mentioned above.

I also noticed another subtle issue when working with `endOf()` or `endOf().minus()`. The dateRange on the top right shows the correct "range", however the `MyFootprint` card suggests that it's a two day range (in the videos, 12/4-12/5).  Upon checking that two-day range, the metrics _are_ the same. 

So,  this current commit "fixed" the WSOD, and made it possible to select one day -- but the behavior isn't exactly what we want, since it technically shows the range of two days.  I think there's a couple things I want to try:
- Make the query _actually_ one day; from basic testing, it doesn't behave like the other queries (where one timestamp is sufficient), so I want to look into how we could pull a single day
- If we _can't_ pull the single day, I want to brainstorm ways to prompt the user to re-select.  It seems disingenuous to leave this as is, since the info displayed isn't exactly what the user asked for.  I remember reading that the react-native calendar doesn't have a way to prevent a single date from being picked, but I bet we could prompt the user another way (Make the blank background say "Please select two or more days!", or something like that...)

I'll spend some time staring at the server code, and report back later!

## Old "Null" Error ↓
https://github.com/JGreenlee/e-mission-phone/assets/98350084/8d8473fc-6438-443d-953c-d8d24244fa97
## Start Date Returns Blank ↓
https://github.com/JGreenlee/e-mission-phone/assets/98350084/ff1a73a0-6c98-4ff2-82fb-4f492d4dadd4
## "Proper" Single Day Select Returns Blank ↓
https://github.com/JGreenlee/e-mission-phone/assets/98350084/629413ba-4b93-40bc-a1ad-c08bdafb1d0e

## Single Day == Two Day ↓
https://github.com/JGreenlee/e-mission-phone/assets/98350084/6d75d9c5-89a3-4817-8671-f3e0551eabdc

Reply:
Just a quick update on this problem:

Below is, I believe, the call trace for fetching / collecting this data. Trying to organize this trace to I can come back to it: each row has the function, and where it's called (as such, the next row down has the location of the called function's definition).  First two calls are from the client, the rest are from the server.

| Function | Func Call Location ↓ | Notes |
|--------------|------------------|-----|
| GetMetrics() | `MetricsTab.jsx `  | Metrics are sorted as a state |
| "BEMServer fetch"  |  `commHelper.ts` | Posts to server | 
| summarize_by_timestamp() | `cfc_webapp.py`| '/result/metrics/<time_type>'|
| _group_by_timestamp() |  `metrics.py` | Called as HOF in _call_group_fn() |
| get_data_df() |  `time_grouping.py` | DF also used elsewhere| 
| get_time_series() | `analysis_timeseries_queries.py` | called from get_timeseries_for_user |
| BuiltinTimeSeries() | `abstract_timeseries.py` | Abstract calls base Constructor Call |
| _get_query() | `builtin_timeseries.py` | I believe _this_ is fetches from the DB !|

With all of this laid out, I've been struggling to find exaclty _why_ we aren't able to fetch with a single timestamp.  If I understand the stack correctly, the `builtin_timeseries` objects should be able to hold hold single days.  I've got a lot more digging to do!